src/layer.rs

```diff
@@ -117,7 +117,7 @@ where
         Box::pin(async move {
             let res = res.await.unwrap(); // inner service is infallible
 
-            if res.status().is_success() {
+            if res.status().is_success() || res.status().is_redirection() {
```

Comment:
It might be safer to check that there was no error:
```suggestion
            if !(res.status().is_server_error() || res.status().is_client_error()) {
```